File: dotnet/Vaas/src/Vaas/Options/ForSha256Options.cs
namespace Vaas.Options;

public class ForSha256Options
{
    public bool UseCache { get; init; } = true;
    public bool UseHashLookup { get; init; } = true;

    public string? VaasRequestId { get; init; }

    public static ForSha256Options From(VaasOptions options)
    {
        return new ForSha256Options
        {
            UseCache = options.UseCache,
            UseHashLookup = options.UseHashLookup,
            VaasRequestId = null,
        };
    }
}

File: typescript/src/options/ForSha256Options.ts
import { VaasOptions } from "./VaasOptions";

export class ForSha256Options {
  public useCache: boolean;
  public useHashLookup: boolean;
  public vaasRequestId?: string;

  constructor(options?: {
    useCache?: boolean;
    useHashLookup?: boolean;
    vaasRequestId?: string;
  }) {
    this.useCache = options?.useCache ?? true;
    this.useHashLookup = options?.useHashLookup ?? true;
    this.vaasRequestId = options?.vaasRequestId;
  }

  public static from(options: VaasOptions): ForSha256Options {
    return new ForSha256Options({
      useCache: options.useCache,
      useHashLookup: options.useHashLookup,
    });
  }
}
